From 65d0e47fcede58cb78df8e9a29ff669e2287273b Mon Sep 17 00:00:00 2001
From: gnome-mpv <gnome-mpv@nowhere>
Date: Sun, 21 Nov 2021 17:54:34 +0700
Subject: [PATCH] main-window: Fix window size shrinking across sessions

Fixes #713.
---
 src/celluloid-main-window.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/celluloid-main-window.c b/src/celluloid-main-window.c
index e96b50f3..6af84749 100644
--- a/src/celluloid-main-window.c
+++ b/src/celluloid-main-window.c
@@ -576,8 +576,8 @@ celluloid_main_window_save_state(CelluloidMainWindow *wnd)
 
 	if(!maximized)
 	{
-		g_settings_set_int(settings, "width", width - 52);
-		g_settings_set_int(settings, "height", height - 52);
+		g_settings_set_int(settings, "width", width);
+		g_settings_set_int(settings, "height", height);
 	}
 
 	if(celluloid_main_window_get_playlist_visible(wnd))
